prometheus-connector/src/test/java/org/apache/flink/connector/prometheus/sink/PrometheusStateSerializerTest.java

```diff
@@ -0,0 +1,108 @@
+/*
+ *  Licensed to the Apache Software Foundation (ASF) under one or more
+ *  contributor license agreements.  See the NOTICE file distributed with
+ *  this work for additional information regarding copyright ownership.
+ *  The ASF licenses this file to You under the Apache License, Version 2.0
+ *  (the "License"); you may not use this file except in compliance with
+ *  the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.apache.flink.connector.prometheus.sink;
+
+import org.apache.flink.connector.base.sink.writer.BufferedRequestState;
+import org.apache.flink.connector.base.sink.writer.ElementConverter;
+import org.apache.flink.connector.base.sink.writer.RequestEntryWrapper;
+import org.apache.flink.connector.prometheus.sink.prometheus.Types;
+
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class PrometheusStateSerializerTest {
+
+    private static final ElementConverter<PrometheusTimeSeries, Types.TimeSeries>
+            ELEMENT_CONVERTER = new PrometheusTimeSeriesConverter();
+
+    private static PrometheusTimeSeries getTestTimeSeries(int i) {
+        return PrometheusTimeSeries.builder()
+                .withMetricName("metric-name")
+                .addLabel("dimension-a", "value-" + i)
+                .addSample(i + 42.0, i + 1L)
+                .addSample(i + 3.14, i + 2L)
+                .build();
+    }
+
+    // This method uses the same implementation as PrometheusSinkWriter.getSizeInBytes() to extract
+    // the requestEntry "size"
+    // (i.e. the number of Samples). This is the "size" used in RequestEntryWrapper
+    // see
+    // https://github.com/apache/flink/blob/69e812688b43be9a0c4f79e6af81bc2d1d8a873e/flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/sink/writer/AsyncSinkWriterStateSerializer.java#L60
```

Comment:
Can we consider making this a javadoc + link to code?
For example:
```
    /**
     * This method uses the same implementation as PrometheusSinkWriter.getSizeInBytes() to extract
     * the requestEntry "size"
     * (i.e. the number of Samples). This is the "size" used in RequestEntryWrapper
     * see {@link org.apache.flink.connector.base.sink.writer.AsyncSinkWriterStateSerializer#serialize(BufferedRequestState)}
     */
```

Reply:
(we can replace the qualified name with an import for the class)

Reply:
Replaced with @link to the method.
This way we lose the pointer to the specific line, but okay

Reply:
Reviewing this, I put back the link to the implementation in GitHub.
What we want to show is the implementation, not the javadoc.